### DISCUSSION TITLE
example for animation

### DISCUSSION THREAD

**Discussion by CATdj** [Kitware/trame#315] [Trame/VTK] [answered]
Hi, I'm completely new to VTK and Trame. Trying to modify some of the examples to make some animation using vtkRenderWindowInteractor, but the model will only update on the VTK window not the web, do you have a simple example that I can follow? Thanks. This is what I did:

def callback_func(caller, timer_event):
    state.contour_value += 10
    renderWindow.Render()
    commit_changes()
    ctrl.view_update()
    
def start_anim():
    renderWindowInteractor.CreateRepeatingTimer(500)
    renderWindowInteractor.AddObserver("TimerEvent", callback_func)
    renderWindowInteractor.Start()

**Comment by jourdain** [comment, marked as answer]
With trame, you can not run `renderWindowInteractor.Start()` otherwise, you will keep the server busy.

You have [an example doing animation with ParaView](https://github.com/Kitware/trame/tree/master/examples/07_paraview/TimeAnimation) but the key thing is to rely on asyncio to schedule the next step.

**Reply by CATdj** [reply]
Thank you for your prompt response, I'll check it out
